src/BenchmarkDotNet.Core/Templates/UapBenchmarkProgram.txt

```diff
@@ -0,0 +1,188 @@
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Environments;
+using BenchmarkDotNet.Jobs;
+using System;
+using System.Linq;
+using System.Reflection;
+using Windows.UI.Xaml;
+using Windows.Foundation.Diagnostics;
+using System.Threading.Tasks;
+
+$TargetTypeNamespace$
+$TargetMethodReturnTypeNamespace$
+
+$AdditionalLogic$
+
+namespace BenchmarkDotNet.Autogenerated
+{
+    public static class Program
+    {
+		static void Main(string[] args)
+        {
+            global::Windows.UI.Xaml.Application.Start((p) => new App(args));
+        }
+        }
+		
+        $ExtraAttribute$
+    sealed partial class App : Application
+    {
+        LoggingChannel lc = new LoggingChannel("BenchmarkDotNet.Autogenerated", null, new Guid("4bd2826e-54a1-4ba9-bf63-92b73ea1ac4a"));
+
+        public App(string[] args)
+        {
+            bool isDiagnoserAttached = args != null && args.Contains(BenchmarkDotNet.Engines.Engine.Signals.DiagnoserIsAttachedParam);
+
+            // the first thing to do is to let diagnosers hook in before anything happens
+            // so all jit-related diagnosers can catch first jit compilation!
+            if (isDiagnoserAttached)
+                lc.LogMessage(BenchmarkDotNet.Engines.Engine.Signals.BeforeAnythingElse);
+
+            try
+            {
+                Action action = () =>
+                {
+                    try
+                    {
+                        // we are not using Runnable here in any direct way in order to avoid strong dependency Main<=>Runnable
+                        // which could cause the jitting/assembly loading to happen before we do anything
+                        // we have some jitting diagnosers and we want them to catch all the informations!!
+
+                        Type type = typeof(Program).GetTypeInfo().Assembly.GetType("BenchmarkDotNet.Autogenerated.Runnable");
+
+                        type.GetMethod("Run", BindingFlags.Public | BindingFlags.Static).Invoke(null, new object[] { isDiagnoserAttached, lc });
+                    }
+                    catch (Exception ex)
+                    {
+                        lc.LogMessage(string.Format("// Exception: {0}. StackTrace: {1}", ex.Message, ex.StackTrace));
+                    }
+                    finally
+                    {
+						lc.Dispose();
+                        Application.Current.Exit();
+                    }
+                };
+
+                Task.Factory.StartNew(action);
+            }
+            catch (Exception ex)
```

Comment:
Problem I've mentioned I've fixed with additional page (MainPage.xaml). The benchmark logic runs as background task, and application doesn't crash in release config.

What investigation should I perform to be sure that nothing affects a results?